src/System.Private.CoreLib/src/System/Runtime/MemoryFailPoint.cs

```diff
@@ -301,14 +310,8 @@ LastKnownFreeAddressSpace < (long)segmentSize)
                 CheckForFreeAddressSpace(segmentSize, true);
 
             RuntimeHelpers.PrepareConstrainedRegions();
```

Comment:
If you are deleting the dummy try/finally blocks, you should delete the PrepareConstrainedRegions call too. PrepareConstrainedRegions makes sense only together with the dummy try/finally. It does nothing without it.